File: app/src/main/java/rasel/lunar/launcher/settings/SettingsPrefsUtils.java
/*
 * Lunar Launcher
 * Copyright (C) 2022 Md Rasel Hossain
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package rasel.lunar.launcher.settings;

import android.content.Context;
import android.content.SharedPreferences;

import rasel.lunar.launcher.helpers.Constants;

public class SettingsPrefsUtils {

    private final Constants constants = new Constants();

    protected void saveTimeFormat(Context context, int timeFormatValue) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_TIME_FORMAT, timeFormatValue).apply();
    }

    protected void saveDateFormat(Context context, String dateFormatValue) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putString(constants.SHARED_PREF_DATE_FORMAT, dateFormatValue).apply();
    }

    protected void saveCityName(Context context, String cityName) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putString(constants.SHARED_PREF_CITY_NAME, cityName).apply();
    }

    protected void saveOwmKey(Context context, String owmKey) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putString(constants.SHARED_PREF_OWM_KEY, owmKey).apply();
    }

    protected void saveTempUnit(Context context, int tempUnit) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_TEMP_UNIT, tempUnit).apply();
    }

    protected void showCity(Context context, int showCity) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_SHOW_CITY, showCity).apply();
    }

    protected void showTodos(Context context, int showTodos) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_SHOW_TODOS, showTodos).apply();
    }

    protected void saveFeedUrl(Context context, String feedUrl) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putString(constants.SHARED_PREF_FEED_URL, feedUrl).apply();
    }

    protected void saveLockMode(Context context, int lockMode) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_LOCK, lockMode).apply();
    }

    protected void saveTheme(Context context, int themeValue) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(constants.SHARED_PREFS_SETTINGS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(constants.SHARED_PREF_THEME, themeValue).apply();
    }
}

File: app/src/main/kotlin/rasel/lunar/launcher/settings/SettingsPrefsUtils.kt
/*
 * Lunar Launcher
 * Copyright (C) 2022 Md Rasel Hossain
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package rasel.lunar.launcher.settings

import android.content.Context
import rasel.lunar.launcher.helpers.Constants


internal class SettingsPrefsUtils {

    private val constants = Constants()

    /* save settings value for time format */
    fun saveTimeFormat(context: Context, timeFormat: Int) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putInt(constants.KEY_TIME_FORMAT, timeFormat).apply()
    }

    /* save settings value for date format */
    fun saveDateFormat(context: Context, dateFormat: String) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putString(constants.KEY_DATE_FORMAT, dateFormat).apply()
    }

    /* save city name */
    fun saveCityName(context: Context, cityName: String) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putString(constants.KEY_CITY_NAME, cityName).apply()
    }

    /* save open weather map api key */
    fun saveOwmApi(context: Context, owmApi: String) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putString(constants.KEY_OWM_API, owmApi).apply()
    }

    /* save settings value for temperature unit */
    fun saveTempUnit(context: Context, tempUnit: Int) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putInt(constants.KEY_TEMP_UNIT, tempUnit).apply()
    }

    /* save settings value for show city name with weather */
    fun showCity(context: Context, showCity: Boolean) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putBoolean(constants.KEY_SHOW_CITY, showCity).apply()
    }

    /* save settings value for todo count on home screen */
    fun todoCount(context: Context, todoCount: Int) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putInt(constants.KEY_TODO_COUNTS, todoCount).apply()
    }

    /* save settings value todo manager lock */
    fun todoLock(context: Context, todoLock: Boolean) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putBoolean(constants.KEY_TODO_LOCK, todoLock).apply()
    }

    /* save settings value for search with keyboard */
    fun keyboardSearch(context: Context, keyboardSearch: Boolean) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putBoolean(constants.KEY_KEYBOARD_SEARCH, keyboardSearch).apply()
    }

    /* save rss feed url */
    fun saveRssUrl(context: Context, rssUrl: String) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putString(constants.KEY_RSS_URL, rssUrl).apply()
    }

    /* save settings value for double tap lock method */
    fun saveLockMethod(context: Context, lockMethod: Int) {
        val sharedPreferences = context.getSharedPreferences(constants.PREFS_SETTINGS, 0)
        sharedPreferences.edit().putInt(constants.KEY_LOCK_METHOD, lockMethod).apply()
    }

}
